docs/design/app/workflows.md

```diff
@@ -0,0 +1,15 @@
+### Pre-Handle Workflow
+
+The `prehandle` package contains the workflow for pre-handling transactions. A rough overview can be seen in the diagram below.
+
+![Diagram of pre-handle transaction workflow](images/Pre-Handle%20Transaction%20Workflow.png)
+
+An `Event` at a time is sent to the `prehandle` with a reference to the latest immutable state. It iterates through each transaction and initiates the pre-handle workflow in a separate thread. The workflow consists of the following steps:
+
+0. **Parse Transaction.** The transaction arrives as a byte-array. The required parts are parsed and the common information is validated.
+1. **Call PreTransactionHandler.** Depending on the type of transaction, a specific `PreTransactionHandler` is called. It validates the transaction-specific parts and pre-loads data into the cache. It also creates a `TransactionMetadata` and fills the required keys.
+2. **Check Signature Map.** The signature of all required keys are checked and prepared for verification.
```

Comment:
Instead of **Check Signature Map**, we might say **Build Crypto Sigs from SigMap**.

The idea is that a cryptographic signature has three parts:
  1. Some bytes that are signed; in our case, either the `bodyBytes` for an Ed25519 signature or the Keccak256 hash of the `bodyBytes` for an ECDSA(secp256k1) signature.
  2. An Ed25519 or secp256k1 public key that is supposed to have signed these bytes (these public keys come from e.g. the Hedera key of some `0.0.X` account).
  3. The signature itself---which comes from the `SignatureMap`, based on existence of a unique `SignaturePair` entry whose `pubKeyPrefix` matches the public key in (2).